blog/introducing-api-history.md

```diff
@@ -0,0 +1,269 @@
+> the proposal.
+>
+> This can serve as a single source of truth which can then be used
+> to derive exact version numbers, and no further documentation changes on main
+> are necessary if the change is backported to other branches."
+>
+> — David Sanders [(@dsanders11)][dsanders11] via Slack
+
+We also didn't include removals in the API History, since when an API is removed
+from Electron, it is also removed from the documentation.
+
+### JavaScript details
+
+I originally planned to create a new `@electron/docs-api-history-tools`
+NPM package that would contain scripts for extracting, validating/linting and converting
+the API history in the documentation files.
+
+About a week before the coding period began, and after some discussion with my
+mentors, I realized that was probably unnecessary:
+
+> "Hi everyone, I was thinking about the project after our huddle: Considering
+> that extraction logic will have to be handled differently in `e/website` and
+> `e/lint-roller` because of their dependencies, maybe there is no need for a
+> separate package for API history stuff?"
+>
+> |         Proposed         |        Revised         |
+> | :----------------------: | :--------------------: |
+> | ![proposed][js-proposed] | ![revised][js-revised] |
+>
+> — Piotr Płaczek (me) via Slack
+
+We ultimately decided to not go ahead with my original idea:
+
+> "@Piotr Płaczek that seems fine to me! I think we can always refactor out to a
+> separate module in a later iteration if we find that we need to share a lot of
+> code between the two implementations anyways :slightly_smiling_face:"
+>
+> — Erick Zhao ([@erickzhao][erickzhao]) via Slack
+
+Instead, we divided those various tools across the Electron repos that were most
+relevant to them:
+
+- `yaml-api-history-schema.json`
+  - -> `electron/electron` (`api-history.schema.json`)
+- `lint-yaml-api-history.ts`
+  - -> `electron/lint-roller` (`lint-markdown-api-history.ts`)
+- `extract-yaml-api-history.ts`
+  - -> `electron/website` (`preprocess-api-history.ts`)
+- `yaml-api-history-to-markdown.ts`
+  - -> `electron/website` (`transformers/api-history.ts`)
+  - -> `electron/website` (`ApiHistoryTable.tsx`)
+
+### UI and styling for Electron documentation website
+
+I originally proposed a simple table to display the API History data:
+
+|      Design Prototype (Closed)      |      Design Prototype (Open)      |
+| :---------------------------------: | :-------------------------------: |
+| ![demo1][prototype-closed-proposed] | ![demo2][prototype-open-proposed] |
+
+This is what the final implemented design looks like:
+
+![demo3][open-implemented]
+
+Pretty much the same as the prototype. The most significant addition is the use
+of [SemVer](https://semver.org/) ranges, which were chosen to better communicate
+which versions a feature is present in (thanks Samuel Attard
+[(@MarshallOfSound)](https://github.com/MarshallOfSound) for the suggestion!).
```

Comment:
I
```suggestion
Pretty much the same as the prototype. The most significant addition is the use
of [SemVer](https://semver.org/) ranges, which were chosen to better communicate
which versions a feature is present in (thanks Samuel Attard
[(@MarshallOfSound)](https://github.com/MarshallOfSound) for the suggestion!).

This is important because changes are frequently backported across supported
Electron release lines e.g. a fix may make it into Electron v32.0.0, v31.1.0 and v30.2.0.
This means it is not present in v31.0.0 which a user might mistakenly deduce based
on the fact it is present in a v30.x.x release.
```